src/models/orphanedData.js

```diff
@@ -10,39 +11,49 @@ export default {
   },
   subscriptions: {
     setup({ dispatch, history }) {
-      history.listen(location => {
-        if (enableQueryData(location.pathname, 'orphanedData')) {
-          dispatch({
-            type: 'query',
-            payload: location.pathname.startsWith('/orphanedData') ? queryString.parse(location.search) : {},
-          })
-        }
+      history.listen(({ pathname = '', search = '', hash = '' }) => {
+        if (!enableQueryData(pathname, 'orphanedData') || !hash) return
+
+        const payload = pathname.startsWith('/orphanedData')
+          ? { ...queryString.parse(search), hash }
+          : {}
+
+        dispatch({ type: 'query', payload })
       })
     },
   },
   effects: {
     *query({
-      payload,
+      payload = {},
     }, { call, put }) {
       let { data } = yield call(query)
-      // Front-end filtering
+      const currentOrphanType = payload.hash?.substring(1)
+
+      // filter by orphan type
+      data = data.filter(orphan => {
+        if (currentOrphanType === ORPHAN_TYPES.REPLICA) return orphan.orphanType === ORPHAN_TYPES.REPLICA
+        return orphan.orphanType !== ORPHAN_TYPES.REPLICA
+      })
```

Comment:
Should we explicitly compare the type with each 3 known orphan types?

Reply:
Sure, update in commit https://github.com/longhorn/longhorn-ui/pull/905/commits/3f8c9b85a4188b37b4eba071a0c87d75f31d2a88.